vllm/config.py

```diff
@@ -2909,31 +2908,30 @@ def init_backend(self, vllm_config: "VllmConfig") -> Union[str, Callable]:
         from vllm.compilation.backends import VllmBackend
         return VllmBackend(vllm_config)
 
-    def init_with_cudagraph_sizes(self, sizes_to_specialize: List[int]):
+    def init_with_cudagraph_sizes(self,
+                                  cudagraph_capture_sizes: List[int]) -> None:
         """To complete the initialization of config,
         we need to know the cudagraph sizes."""
 
-        if self.cudagraph_capture_sizes is None:
-            self.capture_sizes = sizes_to_specialize
+        if self.capture_sizes is None:
+            self.capture_sizes = cudagraph_capture_sizes
         else:
-            self.capture_sizes = self.cudagraph_capture_sizes
             logger.info(("cudagraph sizes specified by model runner"
                          " %s is overridden by config %s"),
-                        sizes_to_specialize, self.cudagraph_capture_sizes)
-
-        if self.candidate_compile_sizes is None:
-            self.candidate_compile_sizes = []
-        self.compile_sizes = [
-            x for x in self.candidate_compile_sizes if x in self.capture_sizes
-        ]
-        ignored_sizes = [
-            x for x in self.candidate_compile_sizes
-            if x not in self.capture_sizes
-        ]
-        if ignored_sizes:
-            logger.warning(("candidate_compile_sizes %s are ignored "
-                            "because they are not cudagraph capture sizes."),
-                           ignored_sizes)
+                        cudagraph_capture_sizes, self.capture_sizes)
+
+        computed_compile_sizes = []
+        if self.compile_sizes is not None:
+            for x in self.compile_sizes:
+                if isinstance(x, str):
+                    assert x == "cudagraph", \
+                    "Unrecognized size type in compile_sizes, " \
+                    f"expect 'cudagraph', got {x}"
```

Comment:
Maybe we should be more explicit and name this `cudagraph_capture_sizes`, just to make the cli args more clear. In your example, it wasn't immediately clear what "cudagraph" meant on first glance:
```
vllm serve meta-llama/Meta-Llama-3-8B -O "{'level': 3, 'compile_sizes':['cudagraph', 360], 'capture_sizes': [1, 2, 4]}"
```

Reply:
Or maybe it's clearer like 

```
vllm serve meta-llama/Meta-Llama-3-8B -O "{'level': 3, 'compile_sizes':['capture_sizes', 360], 'capture_sizes': [1, 2, 4]}"
```

Reply:
sounds good, renamed to `cudagraph_capture_sizes` in [29b08b7](https://github.com/vllm-project/vllm/pull/12243/commits/29b08b7838588755b566486ab3a67a2bae23f65c)